src/bci_build/package.py

```diff
@@ -1423,6 +1424,160 @@ def _get_openjdk_kwargs(
 )
 
 
+@enum.unique
+class PhpVariant(enum.Enum):
+    cli = "php"
+    apache = "php-apache"
+    fpm = "php-fpm"
+
+    def __str__(self) -> str:
+        return str(self.value)
+
+
+def _php_entrypoint(variant: PhpVariant) -> str:
+    cmd: str = {
+        PhpVariant.cli: "php",
+        PhpVariant.apache: "apache2-foreground",
+        PhpVariant.fpm: "php-fpm",
+    }[variant]
+    return f"""#!/bin/sh
+set -e
+
+# first arg is `-f` or `--some-option`
+if [ "${{1#-}}" != "$1" ]; then
+	set -- {cmd} "$@"
+fi
+
+exec "$@"
+"""
+
+
+_EMPTY_SCRIPT = """#!/bin/sh
+echo "This script is not required in the PHP SLE BCI."
+"""
+
+
+def _create_php_bci(
+    os_version: OsVersion, php_variant: PhpVariant, php_version: int
+) -> LanguageStackContainer:
+    common_end = "RUN chmod +x /usr/local/bin/docker-*"
+
+    if php_variant == PhpVariant.apache:
+        extra_pkgs = [f"apache2-mod_php{php_version}"]
+        extra_env = {
+            "APACHE_CONFDIR": "/etc/apache2",
+            "APACHE_ENVVARS": "/usr/sbin/envvars",
+        }
+        cmd = ["apache2-foreground"]
+        custom_end = (
+            common_end
+            + """STOPSIGNAL SIGWINCH
+
+# create our own apache2-foreground from the systemd startup script
+RUN cat /usr/sbin/start_apache2 | sed 's|^exec $apache_bin|exec $apache_bin -DFOREGROUND|' > /usr/local/bin/apache2-foreground
```

Comment:
I wonder if we can do that by setting /etc/sysconfig/apache2 define. would have a slightly better optics